### PR TITLE
Issue#140

### PR DESCRIPTION
**Pull request by paul-lemon**
This is a pull request to resolve issues noted in https://github.com/ljay79/jira-tools/issues/140

customFields.gs contained duplicated code for fetching all the fields from the JIRA server.
The work should have been simple but I needed to deal with a few issues

### CODE REVIEW

File: src/customFields.gs [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-// storage of custom Jira field 'EPIC'
-var fieldEpic = {
-  usable:    false,  // true|false
-  key:       'jst_epic',
-  name:      'Epic',
-  link_key:  null, // customfield_10003
-  label_key: null  // customfield_10005
-};
-
-
```
Comment (paul-lemon) [inline src/customFields.gs]: The management of storing the EpicField has been moved into a single model class

File: src/customFields.gs [hunk 2]
```diff
@@ -46,44 +43,11 @@ function getCustomFields( format ) {
...
-        }
-
-        return {
-          key:        cField.key || cField.id, // Server API returns ".id" only while Cloud returns both with same value
-          name:       cField.name,
-          custom:     cField.custom,
-          schemaType: _type,
-          supported:  (arrSupportedTypes.indexOf(_type) > -1)
-        };
-      }) )
```
Comment (paul-lemon) [inline src/customFields.gs]: This is the big chunk of code that was duplicated in getAllJiraFields in jiraCommon

File: src/dialogCustomFields.html [hunk 3]
```diff
@@ -84,7 +84,7 @@
           chkboxInput.setAttribute('value', JSON.stringify(field));
           chkboxLabel.setAttribute('for', 'customfield-id-'+field.key);
           chkboxLabel.innerHTML = field.name;
-          if (!field.supported) chkboxLabel.innerHTML += ' <small><em>(Type: ' + JSON.stringify(field.type) + (field.supported?'':' - not supported') + ')</em></small>';
+          if (!field.supported) chkboxLabel.innerHTML += ' <small><em>(Type: ' + JSON.stringify(field.schemaType) + (field.supported?'':' - not supported') + ')</em></small>';
```
Comment (paul-lemon) [inline src/dialogCustomFields.html]: The rest of the code when fetching fields from JIRA uses schemaType as the property name so I changed it here so I could re-use the common code.

File: src/customFields.gs (hunk 1 above)
Comment (paul-lemon) [reply src/customFields.gs]: As I work more on other issues - I think a model that looks after all Field Definitions may be a good idea. But thats another piece of work as we move the code base to the structure we discussed in the other Github issue

**Comment by ljay79** [comment]
All unit tests executed manually and have passed.
Manual tests in sheets passed.

**Pull request merged**